src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -167,16 +181,281 @@ def _visit_deref(self, node: gtir.FunCall) -> MemletExpr | ValueExpr:
+
+    def _make_unstructured_shift(
+        self,
+        it: IteratorExpr,
+        connectivity: gtx_common.Connectivity,
+        offset_table_node: dace.nodes.AccessNode,
+        offset_expr: IteratorIndexExpr,
+    ) -> IteratorExpr:
+        """Implements shift in unstructured domain by means of a neighbor table."""
+        assert connectivity.neighbor_axis in it.dimensions
+        neighbor_dim = connectivity.neighbor_axis
+        assert neighbor_dim not in it.indices
+
+        origin_dim = connectivity.origin_axis
+        assert origin_dim in it.indices
+        origin_index = it.indices[origin_dim]
+        assert isinstance(origin_index, SymbolExpr)
+
+        if isinstance(offset_expr, SymbolExpr):
+            # use memlet to retrieve the neighbor index
+            shifted_indices = it.indices | {
+                neighbor_dim: MemletExpr(
+                    offset_table_node,
+                    sbs.Indices([origin_index.value, offset_expr.value]),
+                )
+            }
+        else:
+            # dynamic offset: we cannot use a memlet to retrieve the offset value, use a tasklet node
+            dynamic_offset_value = self._make_dynamic_neighbor_offset(
+                offset_expr, offset_table_node, origin_index
+            )
+
+            shifted_indices = it.indices | {neighbor_dim: dynamic_offset_value}
+
+        shifted_indices.pop(origin_dim)
+        return IteratorExpr(it.field, it.dimensions, shifted_indices)
+
+    def _visit_shift(self, node: gtir.FunCall) -> IteratorExpr:
+        shift_node = node.fun
+        assert isinstance(shift_node, gtir.FunCall)
+
+        # here we check the arguments to the `shift` builtin function: the offset provider and the offset value
+        head, tail = self._split_shift_args(shift_node.args)
+        if tail:
+            # we visit a multi-dimensional shift as recursive shift function calls, each returning a new iterator
+            it = self.visit(self._make_shift_for_rest(tail, node.args[0]))
+        else:
+            # the iterator to be shifted is the argument to the function node
+            it = self.visit(node.args[0])
+        assert isinstance(it, IteratorExpr)
+
+        # first argument of the shift node is the offset provider
+        assert isinstance(head[0], gtir.OffsetLiteral)
+        offset = head[0].value
+        assert isinstance(offset, str)
+        offset_provider = self.subgraph_builder.get_offset_provider(offset)
+        # second argument should be the offset value, which could be a symbolic expression or a dynamic offset
+        offset_expr: IteratorIndexExpr
+        if isinstance(head[1], gtir.OffsetLiteral):
+            offset_expr = SymbolExpr(head[1].value, dace.int32)
+        else:
+            dynamic_offset_expr = self.visit(head[1])
+            assert isinstance(dynamic_offset_expr, MemletExpr | ValueExpr)
+            offset_expr = dynamic_offset_expr
```

Comment:
Not really sure if it makes sense, but the current code seems a bit too verbose... 
```suggestion
        offset_expr = SymbolExpr(head[1].value, dace.int32) if isinstance(head[1], gtir.OffsetLiteral) else self.visit(head[1])
            assert isinstance(offset_expr, IteratorIndexExpr)
```